src/SafeAreaContext.tsx

```diff
@@ -59,40 +68,61 @@ export function SafeAreaProvider({
         nativeEvent: { frame: nextFrame, insets: nextInsets },
       } = event;
 
-      if (
-        // Backwards compat with old native code that won't send frame.
-        nextFrame &&
-        (nextFrame.height !== frame.height ||
-          nextFrame.width !== frame.width ||
-          nextFrame.x !== frame.x ||
-          nextFrame.y !== frame.y)
-      ) {
-        setFrame(nextFrame);
-      }
-
-      if (
-        !insets ||
-        nextInsets.bottom !== insets.bottom ||
-        nextInsets.left !== insets.left ||
-        nextInsets.right !== insets.right ||
-        nextInsets.top !== insets.top
-      ) {
-        setInsets(nextInsets);
-      }
+      setFrame((frame) => {
+        if (
+          // Backwards compat with old native code that won't send frame.
+          nextFrame &&
```

Comment:
@janicduplessis was this just so if someone didn't run `pod install` it'd still work?